File: SDK Examples/Examples/C#/DI_StaticDI/StaticDIForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Drawing.Drawing2D;
using Automation.BDaq;

using System.Globalization;

namespace DI_StaticDI
{
   public partial class StaticDIForm : Form
   {
      #region 
      private Label[] m_portNum;
      private Label[] m_portHex;
      private PictureBox[,] m_pictrueBox;
      private const int m_startPort = 0;
      private const int m_portCountShow = 4;
      #endregion

      public StaticDIForm()
      {
          InitializeComponent();
      }

      public StaticDIForm(int deviceNumber)
      {
         InitializeComponent();
         instantDiCtrl1.SelectedDevice = new DeviceInformation(deviceNumber);
      }
      
      private void InstantDiForm_Load(object sender, EventArgs e)
      {
         //The default device of project is demo device, users can choose other devices according to their needs. 
         if (!instantDiCtrl1.Initialized)
          {
             MessageBox.Show("No device be selected or device open failed!", "StaticDI");
              this.Close();
              return;
          }

          this.Text = "Static DI(" + instantDiCtrl1.SelectedDevice.Description + ")";

          m_portNum = new Label[m_portCountShow] { PortNum0, PortNum1, PortNum2, PortNum3 };
          m_portHex = new Label[m_portCountShow] { PortHex0, PortHex1, PortHex2, PortHex3 };
          m_pictrueBox = new PictureBox[m_portCountShow, 8]{
             {pictureBox00, pictureBox01, pictureBox02, pictureBox03, pictureBox04, pictureBox05,pictureBox06, pictureBox07},
             {pictureBox10, pictureBox11, pictureBox12, pictureBox13, pictureBox14, pictureBox15,pictureBox16, pictureBox17},
             {pictureBox20, pictureBox21, pictureBox22, pictureBox23, pictureBox24, pictureBox25,pictureBox26, pictureBox27},
             {pictureBox30, pictureBox31, pictureBox32, pictureBox33, pictureBox34, pictureBox35,pictureBox36, pictureBox37}
          };
          
          //enable the timer to read DI ports status
          timer1.Enabled = true;
      }

      private void timer1_Tick(object sender, EventArgs e)
       {
          // read Di port state
          byte portData = 0;
          ErrorCode err = ErrorCode.Success;

          for (int i = 0; (i + m_startPort) < instantDiCtrl1.Features.PortCount && i < m_portCountShow; ++i)
          {
             err = instantDiCtrl1.Read(i + m_startPort, out portData);
             if (err != ErrorCode.Success)
             {
                timer1.Enabled = false;
                HandleError(err);
                return;
             }

             m_portNum[i].Text = (i + m_startPort).ToString();
             m_portHex[i].Text = portData.ToString("X2");

             // Set picture box state
             for (int j = 0; j < 8; ++j)
             {
                m_pictrueBox[i,j].Image = imageList1.Images[(portData >> j) & 0x1];
                m_pictrueBox[i,j].Invalidate();
             }
          }
       }

      private void HandleError(ErrorCode err)
      {
         if ((err >= ErrorCode.ErrorHandleNotValid) && (err != ErrorCode.Success))
         {
            MessageBox.Show("Sorry ! Some errors happened, the error code is: " + err.ToString(), "Static DI");
         }
      }
   }
}

File: SDK Examples/Examples/VB.Net/DI_StaticDI/StaticDIForm.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports System.Drawing.Drawing2D
Imports Automation.BDaq

Imports System.Globalization

Public Class StaticDIForm
    Inherits Form
#Region ""
    Private m_portNum As Label()
    Private m_portHex As Label()
    Private m_pictrueBox As PictureBox(,)
    Private Const m_startPort As Integer = 0
    Private Const m_portCountShow As Integer = 4
#End Region

    Public Sub New()
        InitializeComponent()
    End Sub

    Public Sub New(ByVal deviceNumber As Integer)
        InitializeComponent()
        InstantDiCtrl1.SelectedDevice = New DeviceInformation(deviceNumber)
    End Sub

    Private Sub StaticDIForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
        'The default device of project is demo device, users can choose other devices according to their needs.
        If Not InstantDiCtrl1.Initialized Then
            MessageBox.Show("Please select a device in control property!", "StaticDI")
            Me.Close()
            Return
        End If

      Me.Text = "Static DI - Run(" + InstantDiCtrl1.SelectedDevice.Description + ")"

        m_portNum = New Label(m_portCountShow - 1) {PortNum0, PortNum1, PortNum2, PortNum3}
        m_portHex = New Label(m_portCountShow - 1) {PortHex0, PortHex1, PortHex2, PortHex3}
        m_pictrueBox = New PictureBox(m_portCountShow - 1, 7) {{pictureBox00, pictureBox01, pictureBox02, pictureBox03, pictureBox04, pictureBox05, _
         pictureBox06, pictureBox07}, {pictureBox10, pictureBox11, pictureBox12, pictureBox13, pictureBox14, pictureBox15, _
         pictureBox16, pictureBox17}, {pictureBox20, pictureBox21, pictureBox22, pictureBox23, pictureBox24, pictureBox25, _
         pictureBox26, pictureBox27}, {pictureBox30, pictureBox31, pictureBox32, pictureBox33, pictureBox34, pictureBox35, _
         pictureBox36, pictureBox37}}

        'enable the timer to read DI ports status
        timer1.Enabled = True
    End Sub

    Private Sub timer1_Tick(ByVal sender As Object, ByVal e As EventArgs) Handles timer1.Tick
        ' read Di port state
        Dim portData As Byte = 0
        Dim err As ErrorCode = ErrorCode.Success

        Dim i As Integer = 0
        While (i + m_startPort) < InstantDiCtrl1.Features.PortCount AndAlso i < m_portCountShow
            err = InstantDiCtrl1.Read(i + m_startPort, portData)
            If err <> ErrorCode.Success Then
                timer1.Enabled = False
                HandleError(err)
                Return
            End If

            m_portNum(i).Text = (i + m_startPort).ToString()
            m_portHex(i).Text = portData.ToString("X2")

            ' Set picture box state
            For j As Integer = 0 To 7
                m_pictrueBox(i, j).Image = imageList1.Images((portData >> j) And &H1)
                m_pictrueBox(i, j).Invalidate()
            Next
            i += 1
        End While
    End Sub

    Private Sub HandleError(ByVal err As ErrorCode)
        If err <> ErrorCode.Success Then
            MessageBox.Show("Sorry ! Some errors happened, the error code is: " & err.ToString(), "StaticDI")
        End If
    End Sub
End Class

